plugins/main/public/components/wz-agent-selector/wz-agent-selector-service.ts

```diff
@@ -0,0 +1,88 @@
+import store from '../../redux/store';
+import { updateCurrentAgentData } from '../../redux/actions/appStateActions';
+import { DATA_SOURCE_FILTER_CONTROLLED_PINNED_AGENT } from '../../../common/constants';
+import { getAngularModule } from '../../kibana-services';
+import { WzRequest } from '../../react-services';
+
+export class PinnedAgentManager {
+  public static NO_AGENT_DATA = {};
+  public static AGENT_ID_VIEW_KEY = 'agent';
+  public static AGENT_ID_URL_VIEW_KEY = 'agentId';
+  public static AGENT_ID_KEY = 'agent.id';
+  public static FILTER_CONTROLLED_PINNED_AGENT_KEY =
+    DATA_SOURCE_FILTER_CONTROLLED_PINNED_AGENT;
+  private AGENT_VIEW_URL = '/agents';
+  private store: any;
+  private location: any;
+  private route: any;
+
+  constructor(inputStore?: any) {
+    this.store = inputStore ?? store;
+    const $injector = getAngularModule().$injector;
+    this.location = $injector.get('$location');
+    this.route = $injector.get('$route');
+  }
+
+  private equalToPinnedAgent(agentData: any): boolean {
+    const pinnedAgent = this.getPinnedAgent();
+    return pinnedAgent?.id === agentData?.id;
+  }
+
+  pinAgent(agentData: any): void {
+    const isPinnedAgent = this.isPinnedAgent();
+    if (isPinnedAgent && this.equalToPinnedAgent(agentData)) {
+      return;
+    }
+    this.store.dispatch(updateCurrentAgentData(agentData));
+    this.location.search(
+      window.location.href.includes(this.AGENT_VIEW_URL)
+        ? PinnedAgentManager.AGENT_ID_VIEW_KEY
+        : PinnedAgentManager.AGENT_ID_URL_VIEW_KEY,
+      String(agentData?.id),
+    );
+    this.route.reload();
+  }
+
+  unPinAgent(): void {
+    this.store.dispatch(
+      updateCurrentAgentData(PinnedAgentManager.NO_AGENT_DATA),
+    );
+    ['agent', 'agentId'].forEach(param => {
```

Comment:
Since it will be centralized, it is not possible to maintain 1?

Reply:
Using only one option instead of using agent and agentId is not desirable to do now because it would involve reviewing all the URLs, so we could do it later